ydb/library/yql/core/cbo/cbo_optimizer_new.h

```diff
@@ -74,6 +79,54 @@ enum EJoinKind: ui32
 EJoinKind ConvertToJoinKind(const TString& joinString);
 TString ConvertToJoinString(const EJoinKind kind);
 
+/**
+ * This is a temporary structure for KQP provider
+ * We will soon be supporting multiple providers and we will need to design
+ * some interfaces to pass provider-specific context to the optimizer
+*/
+struct IProviderContext {
+    virtual ~IProviderContext() = default;
+
+    virtual double ComputeJoinCost(const TOptimizerStatistics& leftStats, const TOptimizerStatistics& rightStats, EJoinImplType joinImpl) const = 0;
```

Comment:
I would rename it to smth like EJoinAlgoType. Don't like this `Impl` in naming.

Reply:
done